lib/internal/test_runner/tap_stream.js

```diff
@@ -187,7 +187,10 @@ function jsToYaml(indent, name, value) {
 
     // If the ERR_TEST_FAILURE came from an error provided by user code,
     // then try to unwrap the original error message and stack.
-    if (code === 'ERR_TEST_FAILURE' && (failureType === kTestCodeFailure || failureType === kHookFailure)) {
+    if (code === 'ERR_TEST_FAILURE' && (
+      failureType === kTestCodeFailure || failureType === kHookFailure ||
```

Comment:
I wonder if we shouldn't move these failure types into a Set and just check if the type is in the Set.